diff.py

```diff
@@ -2188,7 +2245,16 @@ def diff_sameline(old: str, new: str) -> None:
         if ignore_last_field:
             newfields = newfields[:-1]
             oldfields = oldfields[:-1]
+        else:
+            # If the last field has a register suffix, i.e. "0x38(r7)"  we split that part out to make it a separate field
+            newfields = newfields[:-1] + newfields[-1].split("(")
+            oldfields = oldfields[:-1] + oldfields[-1].split("(")
         for nf, of in zip(newfields, oldfields):
+            # If the new field is a match to any symbol case
+            if imm_matches_everything(nf, config.arch):
+                # And the old field had a relocation, then ignore this mismatch
+                if old_line.has_symbol:
+                    continue
```

Comment:
merge these into a single if joined with an `and`, and move it to inside the `if nf != of` check

Reply:
I'll do it, but I'm worried what black will do with it

Reply:
worked out fine